File: src-tauri/src/player/classes.rs
use super::stats::Stats;
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize, Debug)]
pub struct Classes {
    pub name: String,
    pub base_stats: Stats,
    pub skills: String,
    pub fire_resistance: f32,
    pub magic_resistance: f32,
    pub frost_resistance: f32,
    pub lightning_resistance: f32,
    pub starting_weapon_id: i32,
    pub starting_armor_id: i32,
}

File: src/Interfaces/Classes.ts
export interface Classes {
  id: number;
  name: string;
  description: string;
  base_stats: string; // JSON string
  skills: string; // JSON string
  fire_resistance: number;
  magic_resistance: number;
  frost_resistance: number;
  lightning_resistance: number;
  starting_weapon_id: number;
  starting_armor_id: number;

}
